src/mesh/accessors_impl.hh

```diff
@@ -274,16 +273,16 @@ inline Edge Boundary::edge()
 inline ElementAccessor<3> Boundary::element_accessor()
 {
     ASSERT_DBG(is_valid());
-    return boundary_data_->mesh_->element_accessor(boundary_data_->bc_ele_idx_);
+    return boundary_data_->mesh_->get_bc_mesh()->element_accessor(boundary_data_->bc_ele_idx_);
 }
 
 inline Region Boundary::region()
 { 
     return element_accessor().region();
 }
 
-inline Element * Boundary::element()
+inline const Element * Boundary::element()
 {
     ASSERT_DBG(is_valid());
-    return &( boundary_data_->mesh_->element_vec_[boundary_data_->bc_ele_idx_] );
+    return &( boundary_data_->mesh_->get_bc_mesh()->element(boundary_data_->bc_ele_idx_) );
```

Comment:
Please, rename `geT_bc_mesh` to `bc_mesh`. Same for other getters, reduce boilerplate.